diff --git a/litex_boards/targets/xilinx_zcu102.py b/litex_boards/targets/xilinx_zcu102.py
index 57aac75..4f9dd31 100755
--- a/litex_boards/targets/xilinx_zcu102.py
+++ b/litex_boards/targets/xilinx_zcu102.py
@@ -82,14 +82,23 @@ class BaseSoC(SoCCore):
                 sys_clk_freq = sys_clk_freq
             )
 
+        #self.add_rom("bootrom", 0xA0000000, 0x1000000, contents=get_mem_data("/workspace/litex/linux-on-litex-rocket/boot.bin", endianness="little"))  # "little" for the default RISCV architectures selected by litex
+        #self.add_constant("ROM_BOOT_ADDRESS", 0xA0000000)
+
 # Build --------------------------------------------------------------------------------------------
 def main():
     from litex.build.parser import LiteXArgumentParser
     parser = LiteXArgumentParser(platform=xilinx_zcu102.Platform, description="LiteX SoC on ZCU102.")
     parser.add_target_argument("--sys-clk-freq", default=125e6, type=float, help="System clock generator.")
+    sdopts = parser.target_group.add_mutually_exclusive_group()
+    sdopts.add_argument("--with-sdcard",     action="store_true",       help="Enable SDCard support.")
+    sdopts.add_argument("--load-bios",       action="store_true",    help="Load BIOS over Etherbone and reboot SoC")
     args = parser.parse_args()
 
     soc = BaseSoC(sys_clk_freq=args.sys_clk_freq, **parser.soc_argdict)
+    soc.platform.add_extension(xilinx_zcu102._sdcard_pmod_io) # pmod on sdcard
+    if args.with_sdcard:
+        soc.add_sdcard()
     builder = Builder(soc, **parser.builder_argdict)
     if args.build:
         builder.build(**parser.toolchain_argdict)
@@ -97,6 +106,32 @@ def main():
     if args.load:
         prog = soc.platform.create_programmer()
         prog.load_bitstream(builder.get_bitstream_filename(mode="sram"))
+    
+    if args.load_bios:
+        from litex import RemoteClient
+        from litex.soc.integration.common import get_mem_data
+
+        def memwrite(wb, data, base=0x40000000, burst=0xff):
+            for i in range(0, len(data), burst):
+                wb.write(base + 4 * i, data[i:i + burst])
+
+        wb = RemoteClient()
+        wb.open()
+
+        bios_bin = os.path.join(builder.software_dir, "bios", "bios.bin")
+        rom_data = get_mem_data(bios_bin, "little")
+        print(f"Loading BIOS from: {bios_bin} starting at 0x{wb.mems.rom.base:08x} ...")
+
+        print('Stopping CPU')
+        wb.regs.ctrl_reset.write(0b10)  # cpu_rst
+
+        memwrite(wb, rom_data, base=wb.mems.rom.base, burst=1)
+        wb.read(wb.mems.rom.base)
+
+        print('Rebooting CPU')
+        wb.regs.ctrl_reset.write(0)
+
+        wb.close()
 
 if __name__ == "__main__":
-    main()
\ No newline at end of file
+    main()
